src/components/Card/HashCard/index.tsx

```diff
@@ -120,7 +133,7 @@ export default ({
             >
               {!loading && <img src={CopyIcon} alt="copy" />}
             </SimpleButton>
-            {counterpartAddr ? (
+            {counterpartAddr && (
```

Comment:
> Do we have a good reason to use a short circuit expression instead of the ternary?

This is a syntax sugar for the pattern of unable to directly write `if (bool) return node` in jsx.
React will ignore ReactNode that is a boolean value.
I prefer this approach because it is widespread, similar to `[...arr]`.

Reply:
> > Do we have a good reason to use a short circuit expression instead of the ternary?
> 
> This is a syntax sugar for the pattern of unable to directly write `if (bool) return node` in jsx. React will ignore ReactNode that is a boolean value. I prefer this approach because it is widespread, similar to `[...arr]`.

It has an implicit precondition that `condition` should be the type of `boolean`. So it's prone to render other falsy values, e,g. `0`, `NaN`

Reply:
> > > Do we have a good reason to use a short circuit expression instead of the ternary?
> > 
> > 
> > This is a syntax sugar for the pattern of unable to directly write `if (bool) return node` in jsx. React will ignore ReactNode that is a boolean value. I prefer this approach because it is widespread, similar to `[...arr]`.
> 
> It has an implicit precondition that `condition` should be the type of `boolean`. So it's prone to render other falsy values, e,g. `0`, `NaN`

I found it was highlighted in react docs that `not to put numbers on the left side of &&`

![image](https://github.com/Magickbase/ckb-explorer-frontend/assets/7271329/e6cb9c38-fb40-45c0-a850-46d49b56e002)

Ref: https://react.dev/learn/conditional-rendering

Reply:
> It has an implicit precondition that `condition` should be the type of `boolean`. So it's prone to render other falsy values, e,g. `0`, `NaN`

> I found it was highlighted in react docs that `not to put numbers on the left side of &&`

If the developer is not aware of this issue, using ternary syntax or using the `&&` syntax will have the same result.
If they are aware of this issue, the result will still be the same.

Example:
```
const val: null | number = 0
return val && <i>val</i>
return val ? <i>val</i> : null
// correct
return val != null && <i>val</i>
return val != null ? <i>val</i> : null
```

Reply:
> > It has an implicit precondition that `condition` should be the type of `boolean`. So it's prone to render other falsy values, e,g. `0`, `NaN`
> 
> > I found it was highlighted in react docs that `not to put numbers on the left side of &&`
> 
> If the developer is not aware of this issue, using ternary syntax or using the `&&` syntax will have the same result. If they are aware of this issue, the result will still be the same.
> 
> Example:
> 
> ```
> const val: null | number = 0
> return val && <i>val</i>
> return val ? <i>val</i> : null
> // correct
> return val != null && <i>val</i>
> return val != null ? <i>val</i> : null
> ```

I didn't get the example

```
const val: null | number = 0
return val && <i>val</i> // => render 0
return val ? <i>val</i> : null // => render nothing
```

Reply:
> > > It has an implicit precondition that `condition` should be the type of `boolean`. So it's prone to render other falsy values, e,g. `0`, `NaN`
> > 
> > 
> > > I found it was highlighted in react docs that `not to put numbers on the left side of &&`
> > 
> > 
> > If the developer is not aware of this issue, using ternary syntax or using the `&&` syntax will have the same result. If they are aware of this issue, the result will still be the same.
> > Example:
> > ```
> > const val: null | number = 0
> > return val && <i>val</i>
> > return val ? <i>val</i> : null
> > // correct
> > return val != null && <i>val</i>
> > return val != null ? <i>val</i> : null
> > ```
> 
> I didn't get the example
> 
> ```
> const val: null | number = 0
> return val && <i>val</i> // => render 0
> return val ? <i>val</i> : null // => render nothing
> ```

Do you mean `rendering 0` is expected, but `val ? <i>val</i> : null renders nothing`, which is unexpected.

In this case, the `falsy` branch should be `0` instead of `null` because it's expected to render something, `null` is unexpected.

The main advantage the ternary brings is handling unexpected `condition`

Reply:
> return val && <i>val</i> // => render 0

Oh, I made a rookie mistake and forgot that this expression will return val. In that case, the ternary syntax should indeed be used first.